sklearn/metrics/_plot/tests/test_plot_confusion_matrix.py

```diff
@@ -264,3 +263,51 @@ def test_confusion_matrix_text_format(pyplot, data, y_pred, n_classes,
     text_text = np.array([
         t.get_text() for t in disp.text_.ravel()])
     assert_array_equal(expected_text, text_text)
+
+
+@pytest.mark.parametrize("values_format", [None])
+def test_confusion_matrix_standard_format(pyplot, data, y_pred, n_classes,
+                                          fitted_clf, values_format):
+
+    def test_function_format(x):
+        if (isinstance(x, float) or x >= 1e7):
+            return '.2g'
+        else:
+            return 'd'
+
+    X, y = data
+
+    cm = np.array([[10000000, 0],
+                   [29, 123123]])
+    disp = plot_confusion_matrix(fitted_clf, X, y,
+                                 include_values=True,
+                                 values_format=values_format)
+    disp.confusion_matrix = np.array([[10000000, 0],
+                                      [29, 123123]])
+    # Plot the values to see if it works
+    disp.plot()
+
+    expected_text = np.array([format(v, test_function_format(v))
```

Comment:
I think we are better off explicitly writing out the expected strings than reproducing the logic from the code. This makes our expectations more explicit and so easier for the reader to confirm that the test is working towards desirable behaviour

Reply:
Are you able to address this, @Rick-Mackenbach? I am very uncomfortable seeing `test_function_format` in the test.

Reply:
Sorry, it seems you did not understand my comment. I mean that `test_function_format` reproduces logic from the implementation. Rather we should be checking the output against expected strings directly.